zone/zone.cpp

```diff
@@ -2949,3 +2949,27 @@ void Zone::LoadDynamicZoneTemplates()
 		dz_template_cache[dz_template.id] = dz_template;
 	}
 }
+
+std::string Zone::GetAAName(int aa_id)
+{
+	if (!aa_id) {
+		return std::string();
+	}
+
+	int current_aa_id = 0;
+
+	for (const auto& r : aa_ranks) {
+		if (r.second.get()->id == aa_id && r.second.get()->base_ability) {
```

Comment:
Should be able to just `.find(aa_id)` instead of looping to find it since it looks like the id is the hash key for `Zone::aa_ranks` unless I'm missing something

Reply:
Changed, good call.